### PR TITLE
ExtruderStack.py: clarify limit_to_extruder logic

### PR DESCRIPTION
**Pull request by digitalfrost**
Simplify and clarify logic linked to limit_to_extruder
in getProperty method.
If statements are now less tangled and duplicated.

### CODE REVIEW

**Review by nallath — APPROVED** [review]

**Comment by nallath** [comment]
This doesn't work. I get the following error: 

Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/jaime/Development/Cura/cura/CuraApplication.py", line 63, in <module>
    from cura.Machines.Models.FavoriteMaterialsModel import FavoriteMaterialsModel
  File "/home/jaime/Development/Cura/cura/Machines/Models/FavoriteMaterialsModel.py", line 4, in <module>
    from cura.Machines.Models.BaseMaterialsModel import BaseMaterialsModel
  File "/home/jaime/Development/Cura/cura/Machines/Models/BaseMaterialsModel.py", line 14, in <module>
    from cura.Settings.CuraContainerRegistry import CuraContainerRegistry
  File "/home/jaime/Development/Cura/cura/Settings/CuraContainerRegistry.py", line 26, in <module>
    from . import ExtruderStack
  File "/home/jaime/Development/Cura/cura/Settings/ExtruderStack.py", line 146
    limit_to_extruder = str(limit_to_extruder) if limit_to_extruder
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: expected 'else' after 'if' expression

**Comment by digitalfrost** [comment]
Sorry, that was a Ruby style if one liner...

**Comment by digitalfrost** [comment]
I think a normal if statement now reads better ...

File: cura/Settings/ExtruderStack.py [hunk 1]
```diff
@@ -141,12 +141,11 @@ def getProperty(self, key: str, property_name: str, context: Optional[PropertyEv
...
-        if limit_to_extruder is not None:
-            if limit_to_extruder == -1:
-                limit_to_extruder = int(cura.CuraApplication.CuraApplication.getInstance().getMachineManager().defaultExtruderPosition)
+        if limit_to_extruder == -1:
+            limit_to_extruder = cura.CuraApplication.CuraApplication.getInstance().getMachineManager().defaultExtruderPosition
+        if limit_to_extruder:
             limit_to_extruder = str(limit_to_extruder)
-
-        if (limit_to_extruder is not None and limit_to_extruder != "-1") and self.getMetaDataEntry("position") != str(limit_to_extruder):
+        if limit_to_extruder != self.getMetaDataEntry("position"):
```
Comment (Joeydelarago) [inline cura/Settings/ExtruderStack.py:148]: This case is entered when limit\_to\_extruder is None, which it did not before.

Also if the defaultExtruderPosition is -1 this case is also entered. I'm not sure how likely this is but it is best to code a bit defensively.

**Review by Joeydelarago — CHANGES_REQUESTED** [review]
I'm getting a recursion error here.

**Comment by saumyaj3** [comment]
Hey @digitalfrost! We believe this issue won't be picked up anytime soon, so I will be deferring the issue so we can focus on the most requested and needed features and bug fixes from our community and from Ultimaker
We are constantly working on making Ultimaker Cura better. Our community is a big part of that by requesting new features and reporting bugs.  So, thanks again, and please let us know if you run into any other issues 💪

**Pull request closed without merging**